internal/security/fileprovider/init.go

```diff
@@ -57,30 +58,39 @@ func (b *Bootstrap) BootstrapHandler(_ context.Context, _ *sync.WaitGroup, _ sta
 	fileOpener := fileioperformer.NewDefaultFileIoPerformer()
 	tokenProvider := authtokenloader.NewAuthTokenLoader(fileOpener)
 
-	var req internal.HttpCaller
-	if caFilePath := cfg.SecretService.CaFilePath; caFilePath != "" {
+	var requester internal.HttpCaller
+	if caFilePath := cfg.SecretStore.CaFilePath; caFilePath != "" {
 		lc.Info("using certificate verification for secret store connection")
 		caReader, err := fileOpener.OpenFileReader(caFilePath, os.O_RDONLY, 0400)
 		if err != nil {
-			lc.Error(fmt.Sprintf("failed to load CA certificate: %s", err.Error()))
+			lc.Errorf("failed to load CA certificate: %s", err.Error())
 			return false
 		}
-		req = secretstoreclient.NewRequestor(lc).WithTLS(caReader, cfg.SecretService.ServerName)
+		requester = pkg.NewRequester(lc).WithTLS(caReader, cfg.SecretStore.ServerName)
 	} else {
 		lc.Info("bypassing certificate verification for secret store connection")
-		req = secretstoreclient.NewRequestor(lc).Insecure()
+		requester = pkg.NewRequester(lc).Insecure()
 	}
-	vaultProtocol := cfg.SecretService.Protocol
-	vaultHost := fmt.Sprintf("%s:%v", cfg.SecretService.Server, cfg.SecretService.Port)
-	vaultClient := secretstoreclient.NewSecretStoreClient(lc, req, vaultProtocol, vaultHost)
 
-	fileProvider := NewTokenProvider(lc, fileOpener, tokenProvider, vaultClient)
+	clientConfig := types.SecretConfig{
+		Type:     secrets.Vault,
+		Host:     cfg.SecretStore.Host,
+		Port:     cfg.SecretStore.Port,
+		Protocol: cfg.SecretStore.Protocol,
+	}
+	client, err := secrets.NewSecretStoreClient(clientConfig, lc, requester)
+	if err != nil {
+		lc.Errorf("error occurred creating SecretStoreClient: %s", err.Error())
+		b.exitCode = 1
```

Comment:
just saw this: if error, should we just exit out instead of proceeding?

Reply:
Yes, it should return ASAP.